source-code/core/app/src/createInlang.ts

```diff
@@ -136,6 +137,23 @@ export const createInlang = async (args: {
 
 		const query = createReactiveQuery(() => messages()!)
 
+		let debouncedSave = (_newMessages) => {
+			// Skip initial call for setup
+			debouncedSave = debounce(500, async (newMessages) => {
+				// console.log('saving changes to messages')
+				try {
+					await resolvedModules()?.data.plugins.data.saveMessages({ messages: newMessages})
```

Comment:
You can use `!` instead of `?` here because we know that this cannot be `undefined` at this point in time.